src/types/viewportManager.ts

```diff
@@ -175,9 +177,24 @@ export function loadFromFile(ctx: GlobalContext) {
     reader.onload = (readerEvent) => {
       const jsonData = readerEvent.target.result as string;
       const graphData: GraphData = JSON.parse(jsonData);
```

Comment:
The `JSON.parse` can also fail (e.g. with an empty file). We can later expand the `try` to include this.